### PR TITLE
add a unique CSS class for every page in the app

### PR DESCRIPTION
**Pull request by aliceriot**
#### What are the relevant tickets?

none

#### What's this PR do?

just adds a unique CSS class (like `.post-page`, `.channel-page`, etc) to each page, so we can do page-specific styling if we want.

#### How should this be manually tested?

Read the code, make sure it makes sense. Check that on each page in the SPA there's a unique class applied to the `.main-content` div.

### CODE REVIEW

**Comment by noisecapella** [comment]
Does the settings page need this too?

File: static/js/hoc/withNavSidebar.js [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
-            <NavSidebar {...this.props} />
-          </Sidebar>
-          <div className="main-content">
-            <WrappedComponent {...this.props} />
+const withNavSidebar = R.curry(
+  (className: string, WrappedComponent: Class<React.Component<*, *>>) => {
+    class WithNavSidebar extends React.Component<*, *> {
+      render() {
+        return (
+          <div className={`content has-left-sidebar ${className}`}>
```
[the remark this answers is not in the document]
Comment (noisecapella) [reply static/js/hoc/withNavSidebar.js]: Or maybe we can just change `AuthRequiredPage` to have it on `content` so that everything is consistent

**Comment by aliceriot** [comment]
Settings page should already be taken care of

File: static/js/hoc/withNavSidebar.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (aliceriot) [reply static/js/hoc/withNavSidebar.js]: I'll change `AuthRequiredPage`

**Pull request merged**